server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -976,31 +966,37 @@ private long balanceTablets() {
             tserverStatusForLevel, partitionedMigrations.get(dl), dl, getTablesForLevel(dl));
         wait = Math.max(tabletBalancer.balance(params), wait);
         long migrationsOutForLevel = 0;
-        for (TabletMigration m : checkMigrationSanity(statusForBalancerLevel.keySet(),
-            params.migrationsOut(), dl)) {
-          final KeyExtent ke = KeyExtent.fromTabletId(m.getTablet());
-          if (partitionedMigrations.get(dl).contains(ke)) {
-            log.warn("balancer requested migration more than once, skipping {}", m);
-            continue;
+        try (
+            var tabletsMutator = getContext().getAmple().conditionallyMutateTablets(result -> {})) {
+          for (TabletMigration m : checkMigrationSanity(statusForBalancerLevel.keySet(),
+              params.migrationsOut(), dl)) {
+            final KeyExtent ke = KeyExtent.fromTabletId(m.getTablet());
+            if (partitionedMigrations.get(dl).contains(ke)) {
+              log.warn("balancer requested migration more than once, skipping {}", m);
+              continue;
+            }
+            migrationsOutForLevel++;
+            tabletsMutator.mutateTablet(ke).requireAbsentOperation()
```

Comment:
Tablets can have a future location, current location, and no location.  We really only want to set migrations when a tablet has a current location.   There could be race conditions where things change while the balance code is running.  We can check the location in the conditional mutation.  Could open a follow on issue for this to add this check.

```suggestion
            tabletsMutator.mutateTablet(ke).requireAbsentOperation().requireAnyCurrentLocation()
```

ideally we only want to set the migration location when the current tablet location differs from the migration location, so could make the check more strict

```suggestion
            tabletsMutator.mutateTablet(ke).requireAbsentOperation().requireCurrentLocationNotEqualTo(m.getNewTabletServer())
```

Reply:
Implemented `requireCurrentLocationNotEqualTo`. Please take a look at the new `CurrentLocationNotEqualToIterator` class